### ISSUE TITLE
Suggestion: add a message when there are no upcoming events

### ISSUE THREAD

**Issue by madcampos** [torontojs/torontojs.com#176]
From a DM on slack:
> I just wanted to suggest one change to the website.  If there are no 'Upcoming Events' some text like ' We don't have upcoming events yet, but if you have any speakers who want to tell us their topics, write to 'some email''.

![image](https://github.com/user-attachments/assets/833df0d0-c730-408e-b52c-b1dab7fbb290)

So, the suggestion is to add a message when there is no upcoming events in the lines of:
> No upcoming events, if you are interested in giving a talk, please reach out to: events@torontojs.com

**Comment by kieran** [comment]
Sounds like a suggestion for @tazsingh

**Comment by madcampos** [comment]
@kieran yeah, I was not sure if the Guild SDK provides a react component or renders things in an iframe. But that sounds like it interacts with the responses from Guild

**Comment by tazsingh** [comment]
I've been summoned!

The solution is to just always have events, duh 🙃 

But actually, our SDK is either a JS SDK (like what TorontoJS is using) or an iframe SDK. With the JS SDK, we can add a callback that's executed when the list is empty. Then the TorontoJS codebase can conditionally render the desired message

What do you think about that as a solution here?

**Comment by madcampos** [comment]
> What do you think about that as a solution for this?

I think it is a good solution to provide the "no events" callback. We can then use it to render something else

**Comment by tazsingh** [comment]
@madcampos @kieran This has now been released in SDK version `0.3.1`!

Just update this from `0.3.0` to `0.3.1` to use the latest version: https://github.com/torontojs/torontojs.com/blob/11ef4841a160704482f0b4f2054fd7742972f78e/index.html#L20

Basically all embeds accept a final `options` argument. In this case, the relevant `renderGuildEventList` embed accepts a 4th argument. The TypeScript type all of the arguments are:

```ts
(element: HTMLElement, guildSlug: string, timeSelection: "UPCOMING" | "PAST", options?: {
    itemGap?: number;
    numberToDisplay?: number;
    expand?: boolean;
    itemsDisplayed?: (itemsDisplayed: number) => unknown;
    hasMore?: (hasMore: boolean) => unknown;
}) => () => void
```

So if you pass a 4th argument here, which is an object, that uses that new `itemsDisplayed` handler. If it returns `0` as `itemsDisplayed`, then you can render a message alternatively! ✨ https://github.com/torontojs/torontojs.com/blob/main/components/guild/index.coffee#L30

I believe that should do it. But let me know if you need anything else 😄

**Comment by kieran** [comment]
we can always tweak the messaging, but this has shipped

**Comment by madcampos** [comment]
Thank you for the swift updates @kieran and @tazsingh! :)

I'm closing this issue now
